eduNEXT/platform-plugin-ontask
# feat: add support for multiple backends and add setting to configure ontask api auth token `AP-1162`

### Description
This PR updates the current integration with OnTask.

The following changes were performed:
1. Update the `create_workflow` endpoint to create an OnTask table.
2. Add support for multiple backends.
3. Update the Completion backend (`CompletionDataSummary`)
4. Add a new Grade backend (`GradeDataSummary`)
5. Add a new Django setting to configure the API Auth Token at the platform level.  
6. Add new unit tests and update the documentation.

### How to Test
Using Tutor:

1. Install this branch in your environment.
2. Install the Tutor plugin of [OnTask](https://github.com/eduNEXT/tutor-contrib-ontask/?tab=readme-ov-file#installation).
3. Add this inline plugin in your Tutor environment. This allows the OnTask tab to be added to the instructor dashboard and enables the option to add custom settings to a course from the advanced settings in Studio.

    ```yml
    name: ontask-settings
    version: 0.1.0
    patches:
      openedx-common-settings: |
        OPEN_EDX_FILTERS_CONFIG = {
            "org.openedx.learning.instructor.dashboard.render.started.v1": {
                "fail_silently": False,
                "pipeline": [
                    "platform_plugin_ontask.extensions.filters.AddInstructorOnTaskTab",
                ]
            },
        }
      common-env-features: |
        ENABLE_OTHER_COURSE_SETTINGS: true
    ```
4. Configure the API Auth Token, you have two alternatives:
    - Add this setting in your LMS settings: `ONTASK_API_AUTH_TOKEN = "your-ontask-api-auth-token"`
    - Go to Advanced Settings from **Studio** > **Settings** > **Advanced Settings** > **Other Course Settings** and add a new key. You can generate the api auth token from **OnTask** > **Profile** >  **API Auth token**

        ```json
        {
            ...
            "ONTASK_API_AUTH_TOKEN": "your-ontask-api-auth-token",
        }
        ```
5. Go to the instructor dashboard in a course from **LMS**  > **Instructor** > **OnTask**.
6. Press the **Create Workflow** button. An OnTask workflow will be created and redirected to it. An OnTask table will also be created for that workflow, you can view it from **Table** > **View data**. **NOTE**: The table will only have for the moment the ID of all the enrollments of the course.
7. Now, click on the **Load Data** button, this will load all the completion and grade (thanks to the support of multiple backends) information for each student in the course. This operation is executed asynchronously.
8. Finally, go to **Table** > **View data** again, you should see all the data loaded.
